storybookjs/storybook
# Docs: Cleanup with Promise instead of setTimeout

## What I did

This PR changes the cleanup of stories in docs mode to be executed with a `Promise`, instead of `setTimeout` as introduced in https://github.com/storybookjs/storybook/pull/21214

When navigating between two docs entries, the `setTimeout` would cause `teardown` of the first set of stories to be called _after_ `renderToCanvas` of the second set of stories. This results in the primary story from the second set of stories getting teared down and you end up with a blank canvas.

Using `Promise.resolve` instead makes sure that `teardown` happen before `renderToCanvas`, while still eliminating the React warning.




## How to test

Navigate between two docs entries and see that there is no blank canvas anymore.



## Checklist



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

      make sure to add the `ci:merged` or `ci:daily` GH label to it.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`
